pkg/controller/replicaset/replica_set.go

```diff
@@ -573,7 +618,6 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	rs, err := rsc.rsLister.ReplicaSets(namespace).Get(name)
 	if errors.IsNotFound(err) {
 		klog.V(4).Infof("%v %v has been deleted", rsc.Kind, key)
-		rsc.expectations.DeleteExpectations(key)
```

Comment:
why not keep this here as well?

Reply:
@liggitt do you see a way this could be reached that isn't covered in update/delete handlers? I'd prefer not to leave dead / unneeded code behind that isn't needed otherwise.

In case some other handler queues the RS in the meantime and it gets processed the worst case I can think of (and very rare) is that it manages to be scheduled at the time when the RS delete handler hasn't been called yet but the RS is already deleted in the store. In that case the sync loop just runs short to return nil on missing RS and right after the expectation is wiped and gets queued by the delete handler to be processed.

Reply:
we're handling the NotFound error here, so I'd expect to handle it completely (by deleting any expectations), not just log/return

Reply:
ok